### PR TITLE
Fix codelabs order in `docs/contribute/`, Enhanced a.html shortcode (with target and rel documentation!), Add `pkam.md` and `cram.md`.

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**- What I did**
1. Fix `spacing.md` `order:`
The order is now:
`button.md` 1
`card.md` 2
`image/index.md` 3
`spacing.md` 4

2. Add `links.md` to `docs/contribute/shortcodes` (See 547f59005267456c5d6181948476fd1905dca6a9)
3. Added target attribute to `a.html` shortcode (See `layouts/shortcodes/a.html`)

**- How to verify it**
image

### CODE REVIEW

File: content/docs/contribute/shortcodes/links.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+layout: codelab
+title: Links
+description: Links shortcodes used in atsign.dev
+draft: false
+order: 5
+---
+
```
Comment (xavierchanth) [inline content/docs/contribute/shortcodes/links.md:8]: Put a line break here like the other pages i.e. `{{<br>}}`

File: content/docs/contribute/shortcodes/links.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+### Links
```
Comment (xavierchanth) [inline content/docs/contribute/shortcodes/links.md]: Keep this title relevant and specific to the content it represents, such as `Anchor Tag (a)`

**Review by xavierchanth — CHANGES_REQUESTED** [review]

File: content/docs/contribute/shortcodes/links.md (hunk 2 above)
Comment (JeremyTubongbanua) [reply content/docs/contribute/shortcodes/links.md]: See 48949faed34b462f06702e2904f11d40d729cc92

File: content/docs/contribute/shortcodes/links.md (hunk 1 above)
Comment (JeremyTubongbanua) [reply content/docs/contribute/shortcodes/links.md:8]: Ah, thank you.

See 48949faed34b462f06702e2904f11d40d729cc92

**Comment by JeremyTubongbanua** [comment]
**New Content**

- `cram.md` 1d5d831f02ceb6b7363079c1b6ccc22f233e7e6d
- `pkam.md` 9c67e4119a8077842cc5531bd62d097a911e3f25

Rough drafts. ^^
Please give feedback on content! @XavierChanth @TylerTrott 

Will seek further proof reading once technical content is good to go.

Preview of `cram.md`:
image

File: content/docs/reference/pkam.md [hunk 3]
```diff
@@ -9,8 +9,38 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
+
+The PKAM private key used by the client to authenticate into their own secondary server. It is just like the CRAM secret except it is now owned by you. 
+
+## @protocol
+
+In this process, a {{< a target="_self" href="/docs/reference/pkam" >}} PKAM {{< /a >}} keypair was generated for subsequent logins to your secondary. PKAM keys will be generated by your device (which is an RSA public/private keypair). Then, a `.atKeys` file should be generated (and kept safely by users) for subsequent logins in your apps. Since the PKAM keypair will now be used to login to your apps, the CRAM secret is no longer valid for authentication into your secondary server.
+{{< a target="_blank" href="https://pub.dev/documentation/at_server_spec/latest/verbs/Cram-class.html" >}} Learn more on CRAM {{< /a >}}
+
+In the {{< a target="_self" href="/docs/atplatform/specification" >}} @protocol, {{< /a >}} PKAM is a verb.
```
Comment (xavierchanth) [inline content/docs/reference/pkam.md]: No need to say `target="_self"` since the browser will default to that if it is left undefined. You don't need to change, but just something to note.

File: content/docs/reference/pkam.md [hunk 4]
```diff
@@ -9,8 +9,38 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
 weight: 208 # For single pages, lower is first.
 ---
 
-TODO PKAM goes here
+## Definition
+ PKAM stands for public key authentication mechanism. It is a mechanism for encrypting/decrypting data between two parties.
+
+## @platform
+
+The @platform's packages and SDKs help apps generate a RSA PKAM public/private keypair and save this in a `.atKeys` file containing these keys. Since Atsign provided the CRAM secret, this poses an invulnerability (because Atsign essentialy generated your password for you). To oppose this invulnerability, an RSA PKAM keypair is generated for subsequent authentications and the CRAM secret is voided.
```
Comment (xavierchanth) [inline content/docs/reference/pkam.md]: `invulnerability` > `vulnerability`

Comment (xavierchanth) [inline content/docs/reference/pkam.md]: `(because Atsign essentialy generated your password for you). To oppose this invulnerability, an RSA PKAM keypair is generated for subsequent authentications and the CRAM secret is voided.` > `since this is a shared secret between you and Atsign. To avert this, a new RSA PKAM keypair is generated on your device and the CRAM secret is no longer valid. The PKAM keypair is used for all subsequent authentications, and as you are the only one with access to the private key, you are the only one who is able to authenticate.`

File: content/docs/reference/pkam.md [hunk 5]
```diff
@@ -9,8 +9,38 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
+
+In the {{< a target="_self" href="/docs/atplatform/specification" >}} @protocol, {{< /a >}} PKAM is a verb.
+
+The pkam verb is used to authenticate one's own self as an owner of a Secondary Server using a PKI style authentication.
+
+The following regex represents the syntax of the pkam verb: `^pkam:(?<signature>.+$)`
+
+Learn more about the pkam verb {{< a target="_blank" href="https://github.com/atsign-foundation/at_protocol/blob/trunk/specification/at_protocol_specification.md#the-cram-verb" >}} here. {{< /a >}}
+
+## Key Definitions
```
Comment (xavierchanth) [inline content/docs/reference/pkam.md:44]: Either put this at the top or use superscripts to point out that there are footer definitions for these acronymns.
Also, since CRAM is referenced, I would include that in this list.

File: content/docs/reference/cram.md [hunk 6]
```diff
@@ -9,8 +9,37 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
-toc: false # Change this to "true" to show the table of contents
+toc: true # Change this to "true" to show the table of contents
 weight: 207 # For single pages, lower is first.
 ---
 
-TODO CRAM goes here
+## Definition
+CRAM is a term you may have heard of when reading about the @platform. CRAM stands for challenge response authentication mechanism. It is an algorithm/mechanism used in cryptography commonly used for authentication in protocols. 
+
+When users get their {{< a target="_self" href="/docs/reference/atsign" >}} @sign {{< /a >}} from {{< a target="_blank" href="https://atsign.com/" >}} atsign.com, {{< /a >}} they receive their CRAM secret in the form of a QR code. Your CRAM secret is just a long string of characters.
```
Comment (xavierchanth) [inline content/docs/reference/cram.md]: `long string of characters` > `long string of randomly generated characters`

File: content/docs/reference/cram.md [hunk 7]
```diff
@@ -9,8 +9,37 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
+
+
+## @platform
+You use your CRAM secret to activate your @ sign and authenticate into your {{< a target="_self" href="/docs/atplatform/secondary-server" >}} secondary server {{< /a >}} for the first time. Once authenticated, you may add, update, delete, and lookup keys in your own secondary server. 
+
+Since the CRAM secret was generated and given by Atsign, that would be like Atsign generating your password for you. To opposite this, @platform's SDKs and authentication services (such as at_onboarding_flutter) make it easy to generate a `.atKeys` file for subsequent logins to your secondary. This `.atKeys` file is generated by your device and contains the necessary keys to encrypt/decrypt data in your secondary server and the secondary server of other @ signs. The `.atKeys` file is closely related to PKAM which you can read more about {{< a target="_self" href="/docs/reference/pkam" >}} here. {{< /a >}}
```
Comment (xavierchanth) [inline content/docs/reference/cram.md]: `To opposite` > `To oppose`

Comment (xavierchanth) [inline content/docs/reference/cram.md]: Please format these on separate lines, markdown renderers do not render newlines unless two spaces are provided before the newline. This makes it easier to edit & review. Some formatters are incompatible with hugo as they will alter shortcodes, however [this formatter](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) tends to work decently well.

File: content/docs/reference/pkam.md [hunk 8]
```diff
@@ -9,8 +9,38 @@ description: | # SEO Description of the page (Shows in google and atsign.dev sea
...
+
+The PKAM public key is given out to other clients who want to send data to you. The data they send to you is encrypted using your public key and is decrypted using your private key. Read more on asymmetric encryption {{< a target="_self" href="/docs/reference/encryption" >}} here. {{< /a >}}. 
+
+The PKAM private key used by the client to authenticate into their own secondary server. It is just like the CRAM secret except it is now owned by you. 
+
+## @protocol
+
+In this process, a {{< a target="_self" href="/docs/reference/pkam" >}} PKAM {{< /a >}} keypair was generated for subsequent logins to your secondary. PKAM keys will be generated by your device (which is an RSA public/private keypair). Then, a `.atKeys` file should be generated (and kept safely by users) for subsequent logins in your apps. Since the PKAM keypair will now be used to login to your apps, the CRAM secret is no longer valid for authentication into your secondary server.
```
Comment (xavierchanth) [inline content/docs/reference/pkam.md]: Don't link back to the same page, especially if there are very few links on the page, bad for SEO.

File: content/docs/reference/pkam.md (hunk 5 above)
Comment (xavierchanth) [reply content/docs/reference/pkam.md:44]: Referring to the "Key Definitions" section

**Review by Jesseadriano10 — DISMISSED** [review]
Good Stuff!

**Comment by JeremyTubongbanua** [comment]
Will
1. Fix link syntax (shortcode -> markdown)
2. Add calonicals

**Comment by JeremyTubongbanua** [comment]
@XavierChanth 
1. Added `rel` to both the `a.html` shortcode AND the `links.md` Anchor Tag (a) documentation. 2edd6beac28102289329526c7282b4345da068f1
2. Made some links `calonical` and used markdown `[]()` where possible. a1fe0b5cb144aed309ebb395179314c71a816550

File: content/docs/contribute/shortcodes/links.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+| Attribute | Description                                            |
+| --------- | ------------------------------------------------------ |
+| href      | The html href to the content when the link is clicked. |
+| target    | (optional) "_blank" for new tab, "_self" for same frame (default, *use markdown for self links: `[here](www.google.com)`),|
+| rel       | (optional) "calonical" for calonical links
+
+#### Example
+
+{{< a target="_blank" href="https://atsign.com/" >}} atsign.com {{< /a >}}
```
Comment (xavierchanth) [inline content/docs/contribute/shortcodes/links.md]: This should match the code example

File: layouts/shortcodes/a.html [hunk 10]
```diff
@@ -1 +1 @@
-<a href="{{ .Get "href" }}">{{ .Inner }}</a>
\ No newline at end of file
+<a target="{{ .Get "target" }}" rel="{{ .Get "rel" }}" href="{{ .Get "href" }}"> {{ .Inner }} </a>
```
Comment (xavierchanth) [inline layouts/shortcodes/a.html]: Don't put spaces before and after `{{ .Inner }}` in some cases this may render unnecessary whitespace.

**Review by xavierchanth — CHANGES_REQUESTED** [review]
@JeremyTubongbanua I've asked @cconstab to review the content as well. Please see my requested changes, but beyond that I think it is fine.

File: content/docs/reference/pkam.md (hunk 8 above)
Comment (JeremyTubongbanua) [reply content/docs/reference/pkam.md]: I think this was already removed

File: content/docs/reference/cram.md (hunk 7 above)
Comment (JeremyTubongbanua) [reply content/docs/reference/cram.md]: Are you talking about the anchor tag shortcodes?

File: content/docs/reference/pkam.md (hunk 8 above)
Comment (xavierchanth) [reply content/docs/reference/pkam.md]: Ah, I think I had started this comment before you made the changes. It doesn't show for you until I submit my review. Please ignore.

File: content/docs/reference/cram.md (hunk 7 above)
Comment (xavierchanth) [reply content/docs/reference/cram.md]: No just the line in general, it is very long.

```md
This shows up
on one line in rendered markdown
since there are no double spaces ending the lines.
```

Comment (JeremyTubongbanua) [reply content/docs/reference/cram.md]: See fba181d2f0dcfd0f4a85256d8eca4c070fd83a1a

File: content/docs/reference/pkam.md (hunk 5 above)
Comment (JeremyTubongbanua) [reply content/docs/reference/pkam.md:44]: Will work on superscripts in another branch

Comment (JeremyTubongbanua) [reply content/docs/reference/pkam.md:44]: See #58

**Comment by JeremyTubongbanua** [comment]
@cconstab 
Please give feedback on `cram.md`, `pkam.md`, and `namespace.md` in files changed.
Thank you!

**Review by xavierchanth — DISMISSED** [review]

**Review by cconstab — COMMENTED** [review]
Couple of things

mechanism is nor right "method" for the M in CRAM

I also would probably not ask folks to call an API directly instead they can use the onboarding widget in Flutter and the onbarding_cli in dart..

Whilst they both use a API its much easier and safer to use the SDK code base

**Comment by JeremyTubongbanua** [comment]
> Couple of things
> 
> mechanism is nor right "method" for the M in CRAM
> 
> I also would probably not ask folks to call an API directly instead they can use the onboarding widget in Flutter and the onbarding_cli in dart..
> 
> Whilst they both use a API its much easier and safer to use the SDK code base

Thank you for the feedback, Colin!

See changes here ef897b011d26f6e151f84d8e8e75b656f7f37fcd

**Review by xavierchanth — APPROVED** [review]
If @cconstab is happy, I am :)

**Review by xavierchanth — APPROVED** [review]
